utils/services/adminServices.ts

```diff
@@ -0,0 +1,316 @@
+        if (response.status === 200) {
+          const validatedCreatedService = Service.decode(response.value);
+
+          return withValidatedOrInternalError(
+            validatedCreatedService,
+            userService => ResponseSuccessJson(userService)
+          );
+        }
+
+        if (response.status === 401) {
+          return ResponseErrorUnauthorized("Unauthorized", "Unauthorized");
+        }
+      });
+    });
+  };
+
+  /**
+   * Creates a new subscription for a user service
+   * @param email: User's email
+   * @param subscriptionId: The subscriptionId related to a new subscription
+   * @param productNamePayload: The product name related to a new subscription
+   */
+  public readonly createSubscription = (
+    email: EmailString,
+    subscriptionId: ServiceId,
+    productNamePayload: ProductNamePayload
+  ): Promise<
+    // tslint:disable-next-line:max-union-size
+    | IResponseErrorInternal
+    | IResponseErrorForbiddenNotAuthorized
+    | IResponseErrorNotFound
+    | IResponseSuccessJson<Subscription>
+  > => {
+    const client = this.apiClient.getClient();
+    return withCatchAsInternalError(async () => {
+      const validated = await client.createSubscription({
+        email,
+        productNamePayload,
+        subscription_id: subscriptionId
+      });
+      return withValidatedOrInternalError(validated, response => {
+        if (response.status === 200) {
+          const validatedSubscription = Subscription.decode(response.value);
+
+          return withValidatedOrInternalError(
+            validatedSubscription,
+            subscription => ResponseSuccessJson(subscription)
+          );
+        }
+
+        if (response.status === 403) {
+          return ResponseErrorForbiddenNotAuthorized;
+        }
+
+        if (response.status === 404) {
+          return ResponseErrorNotFound(
+            "Not Found",
+            "Resource (User or Product) not found"
+          );
+        }
+      });
+    });
+  };
+
+  /**
+   * Updates an existing user's service.
+   * @param service: the service to be updated
+   */
+  public readonly updateService = (
+    service: Service
+  ): Promise<
+    // tslint:disable-next-line:max-union-size
+    | IResponseErrorInternal
+    | IResponseErrorNotFound
+    | IResponseErrorUnauthorized
+    | IResponseSuccessJson<Service>
+  > => {
+    const client = this.apiClient.getClient();
+    return withCatchAsInternalError(async () => {
+      const validated = await client.updateService({
+        service,
+        service_id: service.service_id
+      });
+      return withValidatedOrInternalError(validated, response => {
+        if (response.status === 200) {
+          const validatedUpdatedService = Service.decode(response.value);
+
+          return withValidatedOrInternalError(
+            validatedUpdatedService,
+            userService => ResponseSuccessJson(userService)
+          );
+        }
+
+        if (response.status === 401) {
```

Comment:
ditto

Reply:
Fixed in [056d4d4](https://github.com/pagopa/io-functions-services/pull/68/commits/056d4d4e462e65c5b18cea6ab55dd2250668a09b)